AdobeStockImage/Model/GetImageList.php

```diff
@@ -104,4 +141,75 @@ public function execute(SearchCriteriaInterface $searchCriteria): AssetSearchRes
             throw new LocalizedException($message, $exception, $exception->getCode());
         }
     }
+
+    /**
+     * Generate Asset object
+     *
+     * @param Document $item
+     *
+     * @return AssetInterface
+     */
+    private function convertDocumentToAsset(Document $item): AssetInterface
+    {
+        //@TODO granulate this to components and make asset part construction possibly in builder
+
+        /** @var AssetInterface $asset */
+        $asset = $this->assetFactory->create();
+
+        $asset->setId($item->getId());
+        $asset->setThumbnailUrl($item->getCustomAttribute('thumbnail_url')->getValue());
```

Comment:
I'd use a multidimensional array containing mapping and simply go through this mapping using a recursive method with a foreach loop.

It's better to place this conversion in AdobeStockClient module as fields are part fo Adobe Stock SDK interface

Reply:
I think that it is a little bit risky to use a recursive mapping because the API can be changed or even separated on several calls or else. If we want to have a mapping we should use some adapter to manage this to be saved from Adone API changes.

Or use components :) Where every part of it is an independent object which can be map/get/iterate whatever else to get the asset data in a needed format.